app/controllers/facilities_controller.rb

```diff
@@ -36,7 +36,8 @@ def index
 
   # GET /facilities/:facility_id
   def show
-    raise ActiveRecord::RecordNotFound unless current_facility && current_facility.is_active?
+    return redirect_to(facilities_path) if current_facility.try(:cross_facility?)
```

Comment:
This line and the next are two spots where I would have liked to have `current_facility` set to a null object `Facility` when there is no facility.